### DISCUSSION TITLE
Unable to use Nested Resource configuration

### DISCUSSION THREAD

**Discussion by dinis-rodrigues** [dagster-io/dagster#21148] [Q&A]
I'm trying to define nested resources to be used in a job and their respective op's. Following the docs I'm doing:

```python
from dagster import ConfigurableResource, Definitions

class RunConfigConstants(ConfigurableResource):
    # Define your configuration schema and methods here
    values: Optional[Dict[str, Optional[Any]]]
    
    def setup_for_execution(self, context: InitResourceContext) -> None:
        print(self.values)

class OtherResource(ConfigurableResource):
    constants: ResourceDependency[RunConfigConstants]

    database: Optional[str]


# Instantiate your constants resource
constants = RunConfigConstants.configure_at_launch()

# Define your resources, passing the constants resource to OtherResource
defs = Definitions(
    resources={
        "constants": constants,
        "other_resource": OtherResource(database="some_value", constants=constants),
    }
)
```

The problem is that when I yield a new run with the following configuration, it does not work
```python
run_config = {
    "resources": {
        "constants": {
            "config": {
                "values": "some_values",
            }
        },
        "other_resource": {
            "config": {
                "database": "my_database",
            }
        },
    }
}

job_function.execute_in_process(run_config=run_config)
```

The constants resource is properly initialized, but the constants dependency in other_resource, the values are empty (None), I think its not catching the config passed from the run_config.

Is this expected behaviour? I need to pass the constants configuration to the other_resource config?

I would like to reuse the configuration of the constants resource, which was possible with the legacy resource definitions, and not defining it again for the other_resource resource? Is that possible?

**Comment by garethbrickman** [comment]
I think this duplicates #21140
